modules/InstallBasicPackageFiles.cmake

```
@@ -615,6 +621,24 @@ ${_compatibility_vars}
   if(DEFINED _IBPF_DEPENDENCIES OR (DEFINED _IBPF_PRIVATE_DEPENDENCIES AND _need_private_deps))
     set(PACKAGE_DEPENDENCIES "#### Expanded from @PACKAGE_DEPENDENCIES@ by install_basic_package_files() ####\n\ninclude(CMakeFindDependencyMacro)\n")
 
+    if (DEFINED _IBPF_OVERRIDE_MODULE_PATH)
+
+      string(APPEND PACKAGE_DEPENDENCIES "set(CMAKE_MODULE_PATH_BK \${CMAKE_MODULE_PATH})\n")
+      set(_overridden_module_path "")
+      foreach(_path ${_IBPF_OVERRIDE_MODULE_PATH})
+
+          if (IS_ABSOLUTE ${_path})
+              set(_absolute_module_path ${_path})
+          else()
+              set(_absolute_module_path "${CMAKE_CURRENT_SOURCE_DIR}/${_path}")
```

This will result in a path that refer to the source directory, that is not desirable.  I think that (when properly documented) if the `OVERRIDE_MODULE_PATH` is not an absolute path, I guess it should refer to `CMAKE_INSTALL_PREFIX` .

---

In this way, we can avoid the `file(RELATIVE_PATH` and just specify the ovveride module path w.r.t. to `${PACKAGE_PREFIX_DIR}` .

---

Anyway, the absolute path is not going to be saved in the file. This is going to be turned into a relative path with respect to the installation prefix in https://github.com/robotology/ycm/blob/3481f4a40c99057e14511a5b924ba35ba291788a/modules/InstallBasicPackageFiles.cmake#L636

This was to allow the user to use this option also with modules located in a path relative to the file in which ``install_basic_package_files`` is called.

---

Yes, but this relative path refers to a location in the source directory, that is not going to be available if just the installation prefix of the library is installed on another machine, or if for any other reason the user deletes the source directory or alter its contents (for example changing branch, if it is a git repo). In general the file installed in the installation prefix should be usable on its own indipendently from the source directory, and that is the reason for which (for example) the headers file are installed, even if they are already available in the source directory.

---

Ok, should I throw a warning, or prevent it completely?

---

If the path is relative, it will be implicitly relative to `CMAKE_INSTALL_PREFIX` . If instead is absolute, indeed we should check that the path is contained in `CMAKE_INSTALL_PREFIX`, and if that is not the case raise an error. If we want to simplify even further, we could just support the argument to be a relative path, and just print an error if the passed path is absolute.

---

To be honest, I don't like too much how things would look on the original ``CMakeLists``, as it is not immediately clear that this path would be relative to the installation prefix. In addition, some options of ``install_basic_package_files`` would assume relative paths w.r.t. the source, other the install. 
For this reason, I would prefer keeping the logic as it is, but adding warnings if the paths do not exist or if it is not contained in the ``CMAKE_INSTALL_PREFIX``. What do you think?

---

Ok! I have slightly preference for an actual error rather then a warning due for "fail early fail fast", but it is not  a strong opinion.

---

I would have loved to have a "Are you sure?" message. I guess I prefer leaving the possibility to the developer to shoot in his foot. Worst case, some dependency won't be found, but it can be workarounded after fixing https://github.com/robotology/ycm/pull/356#discussion_r489335916